File: Tp_ProjetBank/CompteASeuil.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Tp_ProjetBank
{
    class CompteASeuil : Compte
    {
        private double seuil;
        public CompteASeuil() { }
        public CompteASeuil(int numero, double soldeInitial, double seuil)
        {
            SetNumero(numero);
            SetSolde(soldeInitial);
            this.seuil = seuil;
        }

        public double GetSeuil()
        {
            return this.seuil;
        }
        public void SetSeuil(double seuil)
        {
            this.seuil = seuil;
        }

        public override void Retirer(double valeur)
        {
            double aRetirer = GetSolde() - valeur;
            if (aRetirer > seuil)
                SetSolde(aRetirer);
            else
                Console.WriteLine("Tentative de retrait > au seuil");
        }

        public override string ToString()
        {
            return base.ToString() + " : seuil = " + seuil;
        }
    }
}

File: Enonces-Formation/Exo15-JUnit/en15.junit.hsql/src/main/java/fr/banque/CompteASeuil.java
package fr.banque;

/**
 * Ceci est la classe Compte a seuil. <br/>
 * Il n'est pas possible de descendre en dessous du seuil.
 */
public class CompteASeuil extends Compte implements ICompteASeuil {
	private static final long serialVersionUID = 1L;
	private Double seuil;

	/**
	 * Constructeur. <br/>
	 */
	public CompteASeuil() {
		this(null, null, null);
	}

	/**
	 * Constructeur de l'objet. <br>
	 *
	 * @param unNumero
	 *            le numero du compte
	 * @param unSoldeInitial
	 *            le solde initial du compte
	 * @param unSeuil
	 *            un seuil
	 */
	public CompteASeuil(Integer unNumero, Double unSoldeInitial, Double unSeuil) {
		super(unNumero, unSoldeInitial);
		this.setSeuil(unSeuil);
	}

	@Override
	public Double getSeuil() {
		return this.seuil;
	}

	@Override
	public void setSeuil(Double unSeuil) {
		this.seuil = unSeuil;
	}

	@Override
	public String toString() {
		return super.toString() + " Seuil=" + this.getSeuil();
	}

	@Override
	public void retirer(Double unMontant) throws BanqueException {
		double simu = this.getSolde().doubleValue() - unMontant.doubleValue();
		if (simu <= this.getSeuil().doubleValue()) {
			throw new BanqueException("Votre seuil de " + this.getSeuil() + " ne vous permet pas de retirer "
					+ unMontant + " de votre compte " + this.getNumero());
		} else {
			super.retirer(unMontant);
		}
	}
}
